lib/types/sms.d.ts

```diff
@@ -32,14 +32,15 @@ export interface SmsRecipient {
     queuedDateUtc: number | null;
     status: string;
     toNumber: string;
+    statusMessage: string;
 }
 export interface SubmitSmsRequest {
     body: string;
     recipients: Recipient[];
     from: string;
     scheduledDate: Date | null;
-    optOutMessage: boolean;
-    addUnsubscribeLink: boolean;
+    optOutMessage: boolean | false;
```

Comment:
No need to type 'addUnsubscribeLink' and 'optOutMessage' to 'false'. 'boolean' is enough

It this 2 are optional, you can do

`
{
...,
  addUnsubscribeLink?: boolean;
  optOutMessage?: boolean;
}
`



then have it default to false on submitFax method

`async submitFax(
    request: SubmitFaxRequest
  ): Promise<BaseResponse<SubmitFaxResponse>> {
    const files = await Promise.all(
      request.documents.map((document) => this.uploadDocument(document))
    );

    if (files.find((file) => !file.success)) {
      throw new NotifyreError(
        'The document upload process failed',
        400,
        files.map((file) => (file.success ? null : file.message))
      );
    }

    const documents: BaseResponse<GetDocumentStatusResponse>[] =
      await this.pollDocumentsStatus(files.map((file) => file.payload!));

    return this.httpClient.post(`${this.basePath}/send`, {
      templateName: request.templateName,
      faxes: {
        clientReference: request.clientReference,
        files: documents.map((document) => document.payload!.id),
        header: request.header,
        isHighQuality: request.isHighQuality,
        recipients: request.recipients,
        scheduledDate: request.scheduledDate
          ? dateToTimestamp(request.scheduledDate)
          : null,
        sendFrom: request.sendFrom,
        senderID: request.sendFrom,
        subject: request.subject,
        addUnsubscribeLink: request?.addUnsubscribeLink || false,
        optOutMessage: request?.optOutMessage || false
      }
    });
  }`
  
  Also submitFax function request doesn't seem inlucde these 2?

Reply:
i've updated the code based on the review. The optOutMessage and addUnsubscribeLink only applies to submitSMS